runtime/list.go

```diff
@@ -306,6 +306,70 @@ func listNE(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return listCompare(f, toListUnsafe(v), w, NE)
 }
 
+func listIndex(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	expectedTypes := []*Type{ListType, ObjectType, IntType, IntType}
```

Comment:
I think that for the last two parameters, index() accepts any object that implements `__index__`. See for example [strIndex()](https://github.com/google/grumpy/blob/master/runtime/str.go#L238) which calls IndexInt() to pull out an integer index from an arbitrary Object.